File: solutions/codechef/COVID19/COVID19_1.c
// DATE: 07/01/2023, 08:25:26
// PROBLEM NAME: Coronavirus Spread
// PROBLEM URL: https://www.codechef.com/problems/COVID19
// PROBLEM DIFFICULTY RATTING: 1219
// STATUS: accepted
// TIME: 0.00
// MEMORY: 5.2M

#include <stdio.h>
#define MIN(a, b)(a > b ? b : a)
#define MAX(a, b)(a > b ? a : b)

int main(void) {
    int t, n;
    scanf("%d", &t);

    while(t--){
        scanf("%d", &n);
        int x[n], count = 1, min_ = -1, max_ = -1;
        
        for (int i=0; i<n; i++) scanf("%d", &x[i]);
        
        for (int i=0; i<n-1; i++){
            if (x[i+1] - x[i] <= 2) count++;
            else{
                // min
                if (min_ == -1) min_ = count;
                else min_ = MIN(min_, count);

                // max
                if (max_ == -1) max_ = count;
                else max_ = MAX(max_, count);
                
                // reset the count
                count = 1;
            }
            
        }

        min_ = min_ != -1 ? MIN(min_, count) : count;
        max_ = max_ != -1 ? MAX(max_, count) : count;

        printf("%d %d\n", min_, max_);
        
        
        
    }
	return 0;
}

File: solutions/codechef/COVID_19/COVID_19_1.java
// DATE: 03/11/2022, 06:37:42
// PROBLEM NAME: Covid and Theatre Tickets
// PROBLEM URL: https://www.codechef.com/problems/COVID_19
// PROBLEM DIFFICULTY RATTING: 1077
// STATUS: accepted
// TIME: 0.10
// MEMORY: 54.2M

import java.util.*;
import java.lang.*;
import java.io.*;

class Codechef{
	public static void main (String[] args) throws java.lang.Exception{
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		
		while(t-- > 0){
		    int n = sc.nextInt();
		    int m = sc.nextInt();

		    System.out.println(((n%2) + (n/2)) * ((m%2) + (m/2)));
		}
	}
}


